__tests__/app.test.js

```diff
@@ -1,14 +1,74 @@
 const pool = require('../lib/utils/pool');
 const setup = require('../data/setup');
-// const request = require('supertest');
-// const app = require('../lib/app');
+const request = require('supertest');
+const app = require('../lib/app');
+const UserService = require('../lib/services/UserService');
 
-describe('backend-express-template routes', () => {
+const mockUser = {
+  firstName: 'Doug',
+  lastName: 'Dimmadome',
+  email: 'Proprietor@Dome.com',
+  password: 'hat',
+};
+
+const registerAndLogin = async (userProps = {}) => {
+  const password = userProps.password ?? mockUser.password;
+
+  const agent = request.agent(app);
+
+  const user = await UserService.create({ ...mockUser, ...userProps });
+
+  const { email } = user;
+  await agent.post('/api/v1/users/sessions').send({ email, password });
+  return [agent, user];
+};
+
+describe('secret routes', () => {
   beforeEach(() => {
     return setup(pool);
   });
-  it('example test - delete me!', () => {
-    expect(1).toEqual(1);
+  it('DELETE should return 404 error when signed out', async () => {
+    const res = await request(app).get('/api/v1/users');
+    expect(res.body).toEqual({
+      message: 'You have to sign in. YOU HAVE TO!',
+      status: 401,
+    });
+  });
+  it('/POST create a new user', async () => {
+    const res = await request(app).post('/api/v1/users').send(mockUser);
+    const { firstName, lastName, email } = mockUser;
+
+    expect(res.body).toEqual({
+      id: expect.any(String),
+      firstName,
+      lastName,
+      email,
+    });
+  });
+  it('/POST logs in the current user', async () => {
+    const [agent, user] = await registerAndLogin();
+    const me = await agent.get('/api/v1/users/me');
+
+    expect(me.body).toEqual({
+      ...user,
+      exp: expect.any(Number),
+      iat: expect.any(Number),
+    });
+  });
+  it('GET all the secrets (if authenticate)!', async () => {
+    const [agent] = await registerAndLogin();
+    const res = await agent.get('/api/v1/secrets');
+    expect(res.body.length).toBe(2);
```

Comment:
not a great test because as soon as somebody adds a secret the test will fail despite what it's trying to test for still working